docs/subnets/subnet-upgrade.md

```diff
@@ -0,0 +1,138 @@
+
+:::warning
+Upgrades are pretty delicate operations you need to be very careful about because wrong configuration, procedural mistakes or a botched upgrade can halt your subnet or even lead to data loss!
+:::  
+
+## General upgrade considerations
+
+When operating a Subnet, you should always keep in mind that Proof of Stake networks like Avalanche can only make progress if sufficient amount of validating nodes are connected and processing transactions. Each validator on a Subnet is assigned a certain `weight`, which is a numerical value representing the significance of the node in consensus decisions. On the Primary Network, weight is equal to the amount of AVAX staked on the node. On Subnets, weight is currently assigned by the Subnet owners when they issue the transaction [adding a validator](../apis/avalanchego/apis/p-chain.md#platformaddsubnetvalidator) to the Subnet.
+
+Subnets can operate normally only if validators representing 80% or more of the cumulative validator weight is connected. If the amount of connected stake falls close to or below 80% Subnet performance (time to finality) will suffer, and ultimately the Subnet will halt (stop processing transactions).
+
+You as a Subnet operator need to ensure that whatever you do, at least 80% of the validators' cumulative weight is connected and working.
+
+:::info
+When assigning weight to the nodes, always use values greater than 20, as that is the default value of `snow-sample-size` network parameter and thus a minimum cumulative amount that is needed for normal Subnet operation. Recall that a validator's weight can't be changed while it is validating, so take care to use an appropriate value.
+:::
+
+## Upgrading subnet validator nodes
+
+AvalancheGo, the node client that is running the Avalanche validators is under constant and rapid development. New versions come out often (roughly every two weeks), bringing added capabilities, performance improvements or security fixes. Updates are usually optional, but from time to time (much rarely than regular updates) there will be an update that includes a network upgrade. Those upgrades are mandatory, and any node that does not upgrade in time will go offline when the network upgrade activates.
+
+That's why having a node upgrade strategy is pretty important, and you should always strive to run the latest version of the AvalancheGo client.
```

Comment:
```suggestion
That's why having a node upgrade strategy is absolutely vital, and you should always update to the latest AvalancheGo client immediately when it is made available.
```

Reply:
This actually seems a bit out of place here because it conflates updating AvalancheGo with updating their network.

Reply:
I think many people may get confused on this point.